### PR TITLE
fix: recompute data range max size on page size change

### PR DESCRIPTION
**Pull request by vursen**
## Description

The PR makes the connector compute the max size of the client-side data range based on the recent page size. Before, it used to be computed once during initialization.

## Type of change

- [x] Bugfix

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/main/java/com/vaadin/flow/component/combobox/test/ComboBoxClientSideDataRangePage.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        for (int i = 0; i < ITEMS_COUNT; i++) {
+            items.add("Item " + i);
+        }
+        comboBox.setItems(items);
+
+        Input setPageSizeInput = new Input();
+        setPageSizeInput.setId("set-page-size");
+        setPageSizeInput.setPlaceholder("Set page size");
+        setPageSizeInput.addValueChangeListener(event -> {
+            Integer pageSize = Integer.parseInt(event.getValue());
```
Comment (sissbruecker) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/main/java/com/vaadin/flow/component/combobox/test/ComboBoxClientSideDataRangePage.java:42]:
```suggestion
            int pageSize = Integer.parseInt(event.getValue());
```

**Review by sissbruecker — APPROVED** [review]

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/LazyLoadingIT.java [hunk 2]
```diff
@@ -96,85 +96,6 @@ public void selectItem_changeFilter_properlyFilteredItems() {
         assertNotRendered("Item 2");
     }
 
-    @Test
```
Comment (tomivirkki) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/LazyLoadingIT.java:99]: Is it safe to remove these tests? Are all the special / edge cases from the old tests covered in the new tests?

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/LazyLoadingIT.java [hunk 3]
```diff
@@ -96,85 +96,6 @@ public void selectItem_changeFilter_properlyFilteredItems() {
...
-        scrollToItem(stringBox, 1000);
-        waitUntil(e -> getOverlayContents().contains("Item 999"));
-        assertLoadedItemsCount(
-                "Expected the last page to be loaded (50 items).", 50,
-                stringBox);
-        assertRendered("Item 999");
-    }
-
-    @Test
-    public void scrollToEnd_scrollUpwards_pagesLoaded() {
```
Comment (vursen) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/LazyLoadingIT.java:155]: This test is covered by `defaultPageSize_scrollToEnd_scrollUpAndDown_morePagesLoaded_overflowingPagesDiscarded`

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/LazyLoadingIT.java [hunk 4]
```diff
@@ -96,85 +96,6 @@ public void selectItem_changeFilter_properlyFilteredItems() {
...
-        }
-
-        assertLoadedItemsCount(
-                "There should be 500 items after scrolling back to start", 500,
-                stringBox);
-        assertRendered("Item 2");
-    }
-
-    @Test
-    public void openPopup_scrollToEnd_onlyLastPageLoaded() {
```
Comment (vursen) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/LazyLoadingIT.java:144]: This test is covered by `defaultPageSize_scrollToEnd_onlyLastPageLoaded`.

**Review by tomivirkki — APPROVED** [review]

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/LazyLoadingIT.java (hunk 2 above)
Comment (vursen) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/LazyLoadingIT.java:99]: Updated the tests so that they also check how many items are loaded before reaching the limit.

**Review by tomivirkki — APPROVED** [review]
